pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/RoundRobinScheduler.java

```diff
@@ -79,6 +83,9 @@ public RoundRobinScheduler(long releaseTimeoutMs, Supplier<Long> ticker) {
 
   @Override
   public void register(OpChain operatorChain, boolean isNew) {
+    if (_isShutDown) {
```

Comment:
Linking to the previous comment, `isShutDown` is set after the queues are drained so there's still potential for a race condition.

Can we instead use a "State" enum and disallow register if the state is SHUTTING_DOWN?

I think the change of this gap causing an issue is low but best to be safe. Lmk if I am missing something.

Reply:
This class is not thread safe. Please see the java decoration and other implementation. It is caller's responsibility to make thread safe call to this call.

Reply:
+1 on @61yao 's comment here